### PR TITLE
Show a notice when WWD domain is in `pending_dns` state

### PR DESCRIPTION
**Pull request by hambai**
While a domain is in `pending_dns` state the user can't change the contact information again, so we show a notice with instruction on what to do.

The message looks like this:
screen shot 2017-06-23 at 5 16 49 pm

This PR depends on D6115-code

### CODE REVIEW

**Comment by clucasrowlands** [comment]
Hello! Thanks for the ping to review copy.

`Domain is pending contact information update`

I am a bit confused about this message in the orange box. Are we simply saying that a user cannot make any updates to their contact info while their domain is pending? 

OR...are we saying that they need to take action and update their contact info in order to complete the process?

(I think it's the first one, but wanted to clarify.)

---

> This usually happens in an instance but if you did change your name, organization or email address you'll have to confirm the change via email.

What do you mean by "this usually happens in an instance"? 

---

On this notice, it _sounds_ like we want to a) tell them their contact info has not yet been changed and b) instruct them to check their email to confirm the change. Is this accurate, or am I misunderstanding?

Just want to fully understand why this message would appear, what we want to tell the user, and what action we would like them to take. Thanks! :)

**Comment by hambai** [comment]
Hi @clucasrowlands ,

I'll try to explain as thoroughly as possible. So - when a user changes their contact information we have 2 distinct cases that are handled separately:

1. When the user has changed their email or name or organization
2. When the user has not change any of those fields

1. In the first case (when email, name or org are changed) a confirmation email(s) are sent to the user old/new email address and the domain is put in "pending update" status. The domain stays in this status till the user clicks on the confirmation links in the emails sent. If the user doesn't click on the confirmation emails links in 5 days the contact information change is canceled and the domain is put back in it's normal status.

2. In the second case the domain goes briefly (a minute or so) in "pending update" status and then the changes are applied.

While the domain is in "pending update" status no additional contact information changes can be made.

Does this make it more clear?

**Comment by clucasrowlands** [comment]
Thanks for the clarification and extra details. The message in yellow makes sense here.

Here are a few suggestions for the paragraph underneath the yellow message:

> Your change will be reflected shortly. Please note that if you made a change to your name, organization, or email address, you must confirm the change by clicking the confirmation link(s) in the email(s) sent to you. If you have any questions or issues, please refer to our support page or contact support.

> Your contact information will be updated momentarily. Please note that if you made a change to your name, organization, or email address, you must confirm the change by clicking the confirmation link(s) in the email(s) sent to you. If you have any questions or issues, please refer to our support page or contact support.

Happy to offer more copy ideas if needed -- thanks!

File: client/my-sites/upgrades/domain-management/edit-contact-info/pending-whois-update-card.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import support from 'lib/url/support';
+
+function PendingWhoisUpdateCard( { translate } ) {
+	return (
+		<Card className="edit-contact-info__pending-whois-update-card">
+			<Notice
+				status="is-warning"
+				showDismiss={ false }
+				isCompact={ false } >
+				<h2>{ translate( 'Domain is pending contact information update' ) }</h2>
```
Comment (klimeryk) [inline client/my-sites/upgrades/domain-management/edit-contact-info/pending-whois-update-card.jsx]: The `h2` tag is not needed.

File: client/my-sites/upgrades/domain-management/edit-contact-info/pending-whois-update-card.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import Notice from 'components/notice';
+import support from 'lib/url/support';
+
+function PendingWhoisUpdateCard( { translate } ) {
+	return (
+		<Card className="edit-contact-info__pending-whois-update-card">
+			<Notice
+				status="is-warning"
+				showDismiss={ false }
+				isCompact={ false } >
```
Comment (klimeryk) [inline client/my-sites/upgrades/domain-management/edit-contact-info/pending-whois-update-card.jsx]: That's the default, so no need to include it.

File: client/my-sites/upgrades/domain-management/edit-contact-info/pending-whois-update-card.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			</Notice>
+			<p>
+				{ translate(
+					'Your change will be reflected shortly. Please note that if you made a change to your name, ' +
+					'organization, or email address, you must confirm the change by clicking the confirmation ' +
+					'link(s) in the email(s) sent to you. If you have any questions or issues, please refer to our ' +
+					'{{supporta}}support page{{/supporta}} or {{a}}contact support{{/a}}.',
+					{
+						components: {
+							a: <a href={ support.CALYPSO_CONTACT } target="_blank" rel="noopener noreferrer" />,
```
Comment (klimeryk) [inline client/my-sites/upgrades/domain-management/edit-contact-info/pending-whois-update-card.jsx]: The contact link navigates within Calypso, so I don't think it should open in a new tab by default.

**Comment by klimeryk** [comment]
And one small design suggestion: maybe move the notice out of the `Card`? Then it's more consistent with how we present notices in other sections of Domain Management:
screen shot 2017-06-29 at 15 12 31

(I've also removed the `p` wrapping the text in the above screenshot, since that just adds unnecessary margin at the bottom)

File: client/my-sites/upgrades/domain-management/style.scss [hunk 4]
```diff
@@ -243,6 +243,12 @@ input[type=radio].domain-management-list-item__radio {
 	}
 }
 
+.edit-contact-info__pending-whois-update-card {
+	.notice {
+		margin-bottom: 16px;
```
Comment (klimeryk) [inline client/my-sites/upgrades/domain-management/style.scss]: Nice attention to detail - the uneven margin was bugging me too! 😀

**Pull request merged**